apps/intellij/src/main/kotlin/dev/nx/console/mcp/McpServerService.kt

```diff
@@ -0,0 +1,191 @@
+
+        if (!workspaceFile.exists()) {
+            return false
+        }
+
+        try {
+            // Parse the XML and check if McpServerCommand exists
+            val doc = parseXmlFile(workspaceFile)
+            val components = doc.getElementsByTagName("component")
+
+            for (i in 0 until components.length) {
+                val component = components.item(i) as Element
+                val componentName = component.getAttribute("name")
+
+                if (componentName == "McpProjectServerCommands") {
+                    // Check if it has a McpServerCommand child
+                    val serverCommands = component.getElementsByTagName("McpServerCommand")
+                    val serverCommandsCount = serverCommands.length
+                    return serverCommandsCount > 0
+                }
+            }
+
+            return false
+        } catch (e: Exception) {
+            logger.error("[MCP] Error checking if MCP server is setup", e)
+            return false
+        }
+    }
+
+    private fun addMCPServerToWorkspaceXml() {
+        val workspaceFile = getWorkspaceXmlFile()
+
+        if (!workspaceFile.exists()) {
+            logger.warn("[MCP] Workspace file does not exist: ${workspaceFile.absolutePath}")
+            return
+        }
+
+        try {
+            val doc = parseXmlFile(workspaceFile)
+
+            // Check if the component already exists
+            var mcpComponent: Element? = null
+            val components = doc.getElementsByTagName("component")
+
+            for (i in 0 until components.length) {
+                val component = components.item(i) as Element
+                if (component.getAttribute("name") == "McpProjectServerCommands") {
+                    mcpComponent = component
+                    break
+                }
+            }
+
+            // If component doesn't exist, create it
+            if (mcpComponent == null) {
+                mcpComponent = doc.createElement("component")
+                mcpComponent.setAttribute("name", "McpProjectServerCommands")
+                doc.documentElement.appendChild(mcpComponent)
+            }
+
+            // Check if McpServerCommand already exists
+            val serverCommands = mcpComponent!!.getElementsByTagName("McpServerCommand")
+            if (serverCommands.length > 0) {
+                // Command already exists, we don't need to add another one
+                logger.warn("[MCP] Mcp component already exists")
+
+                return
+            }
+
+            // Create McpServerCommand element
+            val mcpServerCommand = doc.createElement("McpServerCommand")
+            mcpComponent.appendChild(mcpServerCommand)
+
+            // Add options
+            addOption(doc, mcpServerCommand, "enabled", "true")
+            addOption(doc, mcpServerCommand, "name", "nx-mcp")
+            addOption(doc, mcpServerCommand, "programPath", "npx")
```

Comment:
For windows, we need to make this `npx.cmd` or else we get errors saying that npx is not a windows command.